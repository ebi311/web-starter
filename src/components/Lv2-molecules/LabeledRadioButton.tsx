import React from 'react';
import InputLabel from '../Lv1-atoms/InputLabel';
import RadioButton from '../Lv1-atoms/RadioButton';
import { styled } from '../Theme';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;

interface IProps {
  className?: string;
  label: string;
  onChange: (value: string) => void;
  value: string;
  name: string;
  selectItems: {
    label: string;
    value: string;
  }[];
}

// eslint-disable-next-line require-jsdoc
const LabeledRadioButton: React.FC<IProps> = props => {
  return (
    <Container className={props.className}>
      <InputLabel>{props.label}</InputLabel>
      <RadioButton
        value={props.value}
        onChange={props.onChange}
        name={props.name}
        selectItems={props.selectItems}
      />
    </Container>
  );
};

export default LabeledRadioButton;
