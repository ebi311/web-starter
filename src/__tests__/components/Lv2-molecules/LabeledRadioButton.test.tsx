import React from 'react';
import LabeledRadioButton from '../../../components/Lv2-molecules/LabeledRadioButton';
import { shallow } from 'enzyme';

describe('LabeledRadioButton', () => {
  const items = [
    {
      label: 'r01',
      value: 'v01',
    },
    {
      label: 'r02',
      value: 'v02',
    },
  ];
  test('success', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <LabeledRadioButton
        label="label001"
        name="name001"
        value="v001"
        selectItems={items}
        onChange={onChange}
      />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});
