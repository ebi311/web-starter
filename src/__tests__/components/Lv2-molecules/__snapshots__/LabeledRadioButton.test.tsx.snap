// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledRadioButton success 1`] = `
<div
  className="sc-AxiKw dIfukv"
>
  <InputLabel>
    label001
  </InputLabel>
  <RadioButton
    name="name001"
    onChange={[MockFunction]}
    selectItems={
      Array [
        Object {
          "label": "r01",
          "value": "v01",
        },
        Object {
          "label": "r02",
          "value": "v02",
        },
      ]
    }
    value="v001"
  />
</div>
`;
